### PR TITLE
new main page add

### PR DESCRIPTION
**Pull request by venky-1826**
new mainpage component created

### CODE REVIEW

File: ui/src/App.js [hunk 1]
```diff
@@ -2,46 +2,70 @@ import "./App.css";
...
 function App() {
-  const [sidebarstatus, setSidebarStatus] = useState(false);
-  const sidebarclass = sidebarstatus ? "sidebar" : "sidebar sidebar-collapse";
-  function handlesidebarstatus() {
-    setSidebarStatus((previousstate) => {
-      return !previousstate;
-    });
+  let userrouter=useHistory()
+const[sidebarstatus,setSidebarStatus] = useState(false);
+const islogdin= useSelector(action=>action.login.islogdin)
```
Comment (TriNitSTechnologies) [inline ui/src/App.js:30]: Change the 'action' to 'state'

File: ui/src/App.js [hunk 2]
```diff
@@ -2,46 +2,70 @@ import "./App.css";
...
+const sidebarclass = sidebarstatus ? 'sidebar': "sidebar sidebar-collapse";
+function handlesidebarstatus(){
+  setSidebarStatus((previousstate)=>{
+    return !previousstate;
+  })
+}
+useEffect(()=>{
+  console.log(islogdin)
```
Comment (TriNitSTechnologies) [inline ui/src/App.js:38]: Remove the console.log

File: ui/src/App.js [hunk 3]
```diff
@@ -2,46 +2,70 @@ import "./App.css";
...
+  if(islogdin){
```
Comment (TriNitSTechnologies) [inline ui/src/App.js:39]: Change the variable name to 'isUserLoggedIn'

File: ui/src/App.js (hunk 1 above)
Comment (venky-1826) [reply ui/src/App.js:30]: done

File: ui/src/App.js [hunk 4]
```diff
@@ -67,6 +100,10 @@ function App() {
             <Route path="/documents">
               <Documents />
             </Route>
+            <Route path="/Home" exact>
```
Comment (TriNitSTechnologies) [inline ui/src/App.js:103]: route name should be small letters  '/home'

File: ui/src/App.js (hunk 2 above)
Comment (venky-1826) [reply ui/src/App.js:38]: done

File: ui/src/Components/Store/LoginSice.js [hunk 5]
```diff
@@ -0,0 +1,45 @@
+import { createAsyncThunk, createSlice } from "@reduxjs/toolkit"
+import axios from "axios"
+let username = '';
+export const logindata  = createAsyncThunk('user/logindata', async (payload)=>{
+    username = payload.username;
+    console.log( "ddd " +payload.username)
```
Comment (TriNitSTechnologies) [inline ui/src/Components/Store/LoginSice.js:6]: Remove console.log statement

File: ui/src/Components/Store/LoginSice.js [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+   return axios.post('https://trinitstechnologies.com/demo/api/v1/user/login',payload)
+
+
+
+
+})
+
+const initialState ={
+    username:"",
+    islogdin:false
```
Comment (TriNitSTechnologies) [inline ui/src/Components/Store/LoginSice.js:16]: Change the variable name to 'isUserLoggedIn'

File: ui/src/Components/Store/LoginSice.js [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+        }
+
+
+
+    },
+    extraReducers(builder) {
+        builder.addCase(logindata.fulfilled, (state, action) => {
+            state.username = username
+            state.islogdin=true
+            console.log("fgg "+state.username);
```
Comment (TriNitSTechnologies) [inline ui/src/Components/Store/LoginSice.js:37]: Remove the console.log

File: ui/src/Components/Store/LoginSice.js [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+        //   return action.payload;
```
Comment (TriNitSTechnologies) [inline ui/src/Components/Store/LoginSice.js:38]: Remove commented code.

File: ui/src/App.js (hunk 3 above)
Comment (venky-1826) [reply ui/src/App.js:39]: done

File: ui/src/App.js (hunk 4 above)
Comment (venky-1826) [reply ui/src/App.js:103]: done

File: ui/src/Components/Store/LoginSice.js (hunk 5 above)
Comment (venky-1826) [reply ui/src/Components/Store/LoginSice.js:6]: done

File: ui/src/Components/Store/LoginSice.js (hunk 6 above)
Comment (venky-1826) [reply ui/src/Components/Store/LoginSice.js:16]: done

File: ui/src/Components/Store/LoginSice.js (hunk 7 above)
Comment (venky-1826) [reply ui/src/Components/Store/LoginSice.js:37]: done

File: ui/src/App.js [hunk 9]
```diff
@@ -2,46 +2,70 @@ import "./App.css";
...
+const sidebarclass = sidebarstatus ? 'sidebar': "sidebar sidebar-collapse";
+function handlesidebarstatus(){
+  setSidebarStatus((previousstate)=>{
+    return !previousstate;
+  })
+}
+useEffect(()=>{
+  
+  if(isUserLoggedIn){
+    userrouter.push("/Home")
```
Comment (TriNitSTechnologies) [inline ui/src/App.js:40]: change the "/home"

File: ui/src/Components/Header/Header.js [hunk 10]
```diff
@@ -2,12 +2,25 @@ import "./Header.css";
...
 import { Link } from "react-router-dom";
 import { FiLogIn } from "react-icons/fi";
+
+
 function Header(props) {
+  let userdispatch=useDispatch()
+  function logoutuser(){
+    userdispatch(loginSlice({
+      username:"",
+      islogdin:false
```
Comment (TriNitSTechnologies) [inline ui/src/Components/Header/Header.js:16]: Update here also

File: ui/src/Components/LoginPage/Login.js [hunk 11]
```diff
@@ -1,11 +1,24 @@
...
-import { Link } from "react-router-dom";
+import { useEffect } from "react";
+import { useDispatch, useSelector } from "react-redux";
+import { Link, useHistory } from "react-router-dom";
 import * as Yup from "yup";
+import { logindata } from "../Store/LoginSlice";
 
 export default function Login() {
+  const dispatch = useDispatch();
+  const islogdin= useSelector(action=>action.login.islogdin)
```
Comment (TriNitSTechnologies) [inline ui/src/Components/LoginPage/Login.js:11]: Change here also.

File: ui/src/Components/LoginPage/Login.js [hunk 12]
```diff
@@ -1,11 +1,24 @@
...
+  
+  useEffect(()=>{
```
Comment (TriNitSTechnologies) [inline ui/src/Components/LoginPage/Login.js:13]: Remove this if not using this useEffect()

File: ui/src/Components/LoginPage/Login.js [hunk 13]
```diff
@@ -17,12 +30,11 @@ export default function Login() {
...
           })}
           onSubmit={(values) => {
-            axios
-              .post(
-                "https://trinitstechnologies.com/demo/api/v1/user/login",
-                values
-              )
-              .then((response) => {});
+
+            console.log(values)
```
Comment (TriNitSTechnologies) [inline ui/src/Components/LoginPage/Login.js:34]: Remove console.log

File: ui/src/Components/MainPage/Main-Body.js [hunk 14]
```diff
@@ -0,0 +1,51 @@
+export default function Mainbody(){
+    return(
+        <div>
+            <div className="border mainform  w-75">
+                <div>
+                <h1 className="text-center">Our best features</h1>
+                <p className="text-center">Below are some of the best features available in SSStockAlerts .</p>
```
Comment (TriNitSTechnologies) [inline ui/src/Components/MainPage/Main-Body.js:7]: Change to 'Payroll management'

**Review by Sujatha-pachikayala — APPROVED** [review]

**Pull request merged**